### DISCUSSION TITLE
Screen stays black with #define E0_DRIVER_TYPE TMC2208 (instead of _STANDALONE)

### DISCUSSION THREAD

**Discussion by yanou-g** [Jyers/Marlin#1288] [Q&A] [answered]
I have soldered the pin PA3 to the UART pin of the TMC2208 according to the following tutorial: ​https://drive.google.com/file/d/15cUf10lMxW4NUHE9qVAlamXmULdrOJxz/view

If  I configure a  fresh Marlin 2.0.8 according to the tutorial it seems to work (did not try printing because I have no Z endstop and I did not have the bltouch configured) I could use the M906 commands though.

If I compile this Jyers build, the screen of My E3V2 will stay black on boot. I have tried the config files in the folder "Jyers's Config" (only changed PA13 to PA3) and also customized my old config files, with the same result.

I added the line: #define TMC_UART_PIN_E PA3
turned off S_CURVE_ACCELERATION
enabled LIN_ADVANCE

If I define the E0 driver as TMC2208 the screen stays black. If I define it as TMC2208_STANDALONE it works again.

What could be the reason for that behaviour?

**Comment by Cal-LF** [comment]
Jyers config is his personal. Try the E3V2 Templates config folder and start with one that is close to your setup.

**Reply by yanou-g** [reply]
Thats what I tried as well. Sorry if that was not clear.
As soon as I set the E0 driver to TMC2208 the screen will stay black. It compiles without errors though.
Maybe there is some other setting in some other file I need to change for it to work? 
When configuring a fresh marlin according to the tutorial I followed there are multiple things that need to be changed. Those files look different in Jyers build though and it seems he already made all the nescessary adjustments.

In the mean time I have configured a fresh Marlin 2.0.8 with linear advance(2.0.9 does give me errors on compiling). It works perfectly after calibtrating the K factor and edges look a lot nicer now.

But I would like all the nice stuff from the Jyers Build back...

**Comment by yanou-g** [comment]
I have downloaded jyers Build 1.3.3 now and it works with linear advance. I guess thats ok for now. Kind of got used to the "Probe Assisted Tramming" from V1.3.5 though.
If someone could give me a hint on how to get the newest version working that would be greatly appreciated.

**Comment by TOTA42** [comment]
That's actually pretty interesting. I was building the 2.0.9.2 straight from Marlin's source and was trying to enable the UART for the E0 stepper with the PA3 pin and got the black screen as well.  I'll try and do the same with 2.0.8 source and see if it's the same.

**Reply by TOTA42** [reply]
Just built the same changes on 2.0.8 and was able to boot with the UART on pin PA3.  Not sure what has happened between 2.0.8 and 2.0.9.1/2.0.9.2

**Reply by TOTA42** [reply]
Well it's something that was introduced before 2.0.9, you can refer to the issue I stuck on Marlin's main branch (https://github.com/MarlinFirmware/Marlin/issues/22905) and they have a workaround at the moment. I'm trying to see if I can get away from libmaple though.

**Reply by alionur54** [reply]
I had success with version Jyers 1.3.4
i used pa4 pa7 pa13 pa14 pins

**Reply by TOTA42** [reply]
That would make sense, Jyers 1.3.4 was prior to the commit that caused this issue.

**Comment by MicroSur** [comment]
It was also difficult for me to find a working simple modification for LA with a config from Jyers, but I found here:
https://github.com/Jyers/Marlin/discussions/707

**Comment by alionur54** [comment]
I have the same problem. I'm getting a black screen with the 1.3.5b build.

**Comment by Jyers** [comment, marked as answer]
Try this with the latest source. It's based of the newest upstream code and I have been testing it with my config, which has LA as you know, and it has been working fine :)

**Reply by alionur54** [reply]
Not fixed, black screen again...

**Reply by Jyers** [reply]
Try using just my config, as I can confirm it is working. That will tell us if it's a firmware issue or a hardware issue

**Reply by alionur54** [reply]
OK, I'll try. But I don't think it's a hardware issue. Because I don't get any error in Jyers1.3.4 and Marlin2.0.8.2 versions.

**Reply by Jyers** [reply]
Also make sure you're compiling from the new branch bugfix-JyersUI

**Reply by alionur54** [reply]
I also tried the "bugfix-JyersUI" and "Upstream_JyersUI" versions.

**Reply by Jyers** [reply]
bugfix-JyersUI is what you should be using

**Reply by TOTA42** [reply]
It's an issue with when Marlin switched from libmaple to the stm32duino, which is why I am suspecting the HAL. Did a git bisect and this was the commit that caused it (d13ffa0). A workaround you could do is to change the environment from STM32F103RET6_creality to STM32F103RET6_creality_maple, which worked for me. I have an issue posted on the upstream and they had me test this workaround.

**Reply by MicroSur** [reply]
Simple solution is to use PA13 pin instead of PA3.

**Reply by alionur54** [reply]
I'm already using pins PA4, PA7, PA13 and PA14.

**Reply by alionur54** [reply]
I tried the current "bugfix-JyersUI"version. I had black screen again with my settings. There is no problem with your settings because there is only E axis UART mode. In my own settings, there is no problem when I only do E axis UART. Even if I use "maple" environment, all drivers are UART, no problem. The problem happens when I put all the drivers in UART mode when compiling versions above 1.3.4.

**Reply by Jyers** [reply]
You should be using maple. It's the only stable environment for the board currently

**Comment by IXoUoXI** [comment]
I know this is an older thread, but I recently ran into the same issue with PC1 and wanted to share what worked for me. I'm not an expert, but after struggling with this on my printer, I found that the issue was due to 'hardcoded' analog inputs instead of the expected pin number reference in variant.h within PlatformIO.

A simple fix was to replace it with:
#define PC1 33 //PIN_A11 // | 33 | A11

I haven't tested it extensively with printing yet, but it successfully got me past the boot screen.

**Reply by vmartos** [reply]
I also can confirm that is the fix. I have been doing [UART mod](file:///C:/Users/V/Downloads/Creality4.2.2_4.2.7_TMC_UART_Mod.pdf) on creality 4.2.2 and got blue screen after flashing.
The thing is that certain pins are set to analog by default (I guess that is what A_ prefix stands for). I simply redefined that in buildroot/share/PlatformIO/variants/MARLIN_F103Rx/variant.h (path depends on the environment you compile for, STM32F103RE_creality in my case), which fixes the issue.

For example
#define PC0  PIN_A10
to
#define PC0  32
